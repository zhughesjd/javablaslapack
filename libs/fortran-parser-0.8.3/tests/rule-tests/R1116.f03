! Test block-data
!      block-data  is  block-data-stmt
!                        [ specification-part ]
!                      end-block-data-stmt
!
!      block-data-stmt  is  BLOCK DATA [ block-data-name ]
!
!      end-block-data-stmt  is  END [ BLOCK DATA [ block-data-name ] ] 
!
! Not tested here: specification-part and block-data-name.

! Include none of the optional parts.
block data 
end

! Include just the optional "BLOCK DATA" part of end-block-data-stmt
block data
end block data

! Include the optional block-data-name
block data foo
end block data foo

! Include the optional specification-part
block data foo
   integer i
end block data foo
