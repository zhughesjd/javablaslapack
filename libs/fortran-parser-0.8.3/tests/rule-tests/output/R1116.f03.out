*******************************************
args[1]: R1116.f03
R1117:block-data-stmt__begin:
R1117:block-data-stmt: id=null
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1118:end-block-data-stmt: id=null
R1116:block-data:
Parser exiting normally
R1117:block-data-stmt__begin:
R1117:block-data-stmt: id=null
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1118:end-block-data-stmt: id=null
R1116:block-data:
Parser exiting normally
R1117:block-data-stmt__begin:
R1117:block-data-stmt: id=[@50,546:548='foo',<209>,21:11]
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1118:end-block-data-stmt: id=[@58,565:567='foo',<209>,22:15]
R1116:block-data:
Parser exiting normally
R1117:block-data-stmt__begin:
R1117:block-data-stmt: id=[@67,623:625='foo',<209>,25:11]
R 403:intrinsic-type-spec: keyword1=[@72,630:636='integer',<61>,26:3] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@74,638:638='i',<209>,26:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 207:declaration-construct:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=1
R1118:end-block-data-stmt: id=[@82,655:657='foo',<209>,27:15]
R1116:block-data:
Parser exiting normally
********************************************
